shell_integration/nautilus/syncstate.py

```diff
@@ -86,9 +87,9 @@ def addListener(self, listener):
     def _connectToSocketServer(self):
         try:
             self._sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-            postfix = "/" + appname + "/socket"  # Should use os.path.join instead
-            sock_file = get_runtime_dir() + postfix
-            print ("Socket: " + sock_file + " <=> " + postfix)
+            postfix = os.path.join(appname, "socket")
```

Comment:
Ah yes, I overlooked the sock_file test, sorry. Well it is difficult to say what was meant the `if`. The original checks if the concat with `get_runtime_dir()` introduces some change, else it is an invalid `sock_file`. Here are examples for the edge cases:

>>> os.path.join("", "foo/bar")
'foo/bar'
>>> os.path.join("/", "foo/bar")
'/foo/bar'
>>> os.path.join("/", "/foo/bar")
'/foo/bar'

I should probably be on the safe side and don't introduce a behaviour change. Can I amend the fix to this commit and force push this branch?